pkg/manager/member/pd_member_manager_test.go

```diff
@@ -1408,3 +1408,141 @@ func TestGetNewPdServiceForTidbCluster(t *testing.T) {
+							Labels:      label.New().Instance(cluster.GetInstanceName()).PD().Labels(),
+						},
+					}
+					podIndexer.Add(pod)
+				}
+				for ordinal := 0; ordinal < 3; ordinal++ {
+					pvc := &corev1.PersistentVolumeClaim{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:        ordinalPVCName(v1alpha1.PDMemberType, controller.PDMemberName(cluster.GetName()), int32(ordinal)),
+							Namespace:   metav1.NamespaceDefault,
+							Annotations: map[string]string{},
+							Labels:      label.New().Instance(cluster.GetInstanceName()).PD().Labels(),
+						},
+					}
+					pvcIndexer.Add(pvc)
+				}
+
+			},
+			pdHealth: &pdapi.HealthInfo{Healths: []pdapi.MemberHealth{
+				{Name: "test-pd-0", MemberID: uint64(1), ClientUrls: []string{"http://test-pd-0:2379"}, Health: false},
+				{Name: "test-pd-1", MemberID: uint64(2), ClientUrls: []string{"http://test-pd-1:2379"}, Health: false},
+			}},
+			err: false,
+			expectTidbClusterFn: func(g *GomegaWithT, tc *v1alpha1.TidbCluster) {
+				g.Expect(tc.Status.PD.NotJoinClusterMembers["test-pd-2"]).NotTo(Equal(nil))
+			},
+		},
+		{
+			name: "clear notJoinClusterMembers when the member join the cluster  ",
+			tcStatusChange: func(cluster *v1alpha1.TidbCluster) {
+				cluster.Status.PD.NotJoinClusterMembers = map[string]v1alpha1.NotJoinClusterMember{
+					"test-pd-0": {
+						PodName:   "test-pd-0",
+						CreatedAt: metav1.Now(),
+					},
+				}
+			},
+			modify: func(cluster *v1alpha1.TidbCluster, podIndexer cache.Indexer, pvcIndexer cache.Indexer) {
+				for ordinal := 0; ordinal < 3; ordinal++ {
+					pod := &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:        ordinalPodName(v1alpha1.PDMemberType, cluster.GetName(), int32(ordinal)),
+							Namespace:   metav1.NamespaceDefault,
+							Annotations: map[string]string{},
+							Labels:      label.New().Instance(cluster.GetInstanceName()).PD().Labels(),
+						},
+					}
+					podIndexer.Add(pod)
+				}
+				for ordinal := 0; ordinal < 3; ordinal++ {
+					pvc := &corev1.PersistentVolumeClaim{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:        ordinalPVCName(v1alpha1.PDMemberType, controller.PDMemberName(cluster.GetName()), int32(ordinal)),
+							Namespace:   metav1.NamespaceDefault,
+							Annotations: map[string]string{},
+							Labels:      label.New().Instance(cluster.GetInstanceName()).PD().Labels(),
+						},
+					}
+					pvcIndexer.Add(pvc)
+				}
+
+			},
+			pdHealth: &pdapi.HealthInfo{Healths: []pdapi.MemberHealth{
+				{Name: "test-pd-0", MemberID: uint64(1), ClientUrls: []string{"http://test-pd-0:2379"}, Health: false},
+				{Name: "test-pd-1", MemberID: uint64(2), ClientUrls: []string{"http://test-pd-1:2379"}, Health: false},
+				{Name: "test-pd-2", MemberID: uint64(2), ClientUrls: []string{"http://test-pd-2:2379"}, Health: false},
+			}},
+			err: false,
+			expectTidbClusterFn: func(g *GomegaWithT, tc *v1alpha1.TidbCluster) {
+				g.Expect(len(tc.Status.PD.NotJoinClusterMembers)).To(Equal(0))
```

Comment:
```suggestion
				g.Expect(tc.Status.PD.NotJoinClusterMembers).To(BeEmpty())
```
Use more descriptive ginkgo matchers could improve clarity